src/esperanza/walletextension.cpp

```diff
@@ -179,8 +179,8 @@ bool WalletExtension::CreateRemoteStakingTransaction(const CRecipient &recipient
     *error_out = "Invalid scriptPubKey for recipient";
     return false;
   }
-  if (type != TX_PUBKEYHASH) {
-    *error_out = "Invalid recipient script: must be P2PKH";
+  if (type != TX_PUBKEYHASH && type != TX_WITNESS_V0_KEYHASH) {
```

Comment:
I think we eventually want to remove `TX_PUBKEYHASH` support for staking transactions right?

Reply:
This script is merely used to provide an address of the remote-staking node. It won't be used for staking. We can remove `TX_PUBKEYHASH` here if we don't want to support legacy Base58 addresses.